src/App.jsx

```diff
@@ -14,33 +16,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABETICAL = 'alphabetical';
+const SORT_FIELD_LENGTH = 'length';
+const SORT_FIELD_REVERSE = 'reverse';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [sortField, setSortField] = useState('');
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const reset = () => {
+    setVisibleGoods(goodsFromServer);
+    setSortField('');
+    // eslint-disable-next-line no-use-before-define
+    setIsReversed(false);
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const sortAlphabetically = () => {
+    const sortedGoods = [...goodsFromServer].sort();
+
+    // eslint-disable-next-line no-use-before-define
+    setVisibleGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setSortField(SORT_FIELD_ALPHABETICAL);
+  };
+
+  const sortByLength = () => {
+    const sortedGoods = [...goodsFromServer].sort(
+      (good1, good2) => good1.length - good2.length,
+    );
+
+    // eslint-disable-next-line no-use-before-define
+    setVisibleGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setSortField(SORT_FIELD_LENGTH);
+  };
```

Comment:
```suggestion
  const sortAlphabetically = () => {
    setSortField(SORT_FIELD_ALPHABETICAL);
  };

  const sortByLength = () => {
    setSortField(SORT_FIELD_LENGTH);
  };
```
you don't need that logic, just move sorting and reversing logic to some function and call it in the component

```
const sortedGoods = getSortedGoods(goodsFromServer, sortValue, isReversed);
```
and reuse it everywhere